my-app/src/App.js

```diff
@@ -46,62 +51,92 @@ constructor(props) {
   this.state = { 
     keyword: null, 
     movie: null,
-    number: 1,
+    results: 1,
     searchStatus: false,
     sort: null, 
   }
-  this.handleChange = this.handleChange.bind(this)
-  this.handleData = this.handleData.bind (this)
-  this.handleNumber = this.handleNumber.bind (this)
-  this.handleSort = this.handleSort.bind (this)
+  this.updateKeyword = this.updateKeyword.bind(this)
+  this.updateMovie = this.updateMovie.bind (this)
+  this.updateNoOfResults = this.updateNoOfResults.bind (this)
+  this.updateSortCriteria = this.updateSortCriteria.bind (this)
+  this.fetchRequest = this.fetchRequest.bind (this)
 }
 
-handleChange(e) {
+/*
+shouldComponentUpdate(nextProps, nextState) {
+  if (nextState.searchStatus === false) {
+    return false;
+  } else {
+    return true; 
+  }
+} */
+
+updateKeyword(e) {
   const keyword = e.target.value
   this.setState({ keyword: keyword });
 }
 
-handleNumber(e) {
+updateNoOfResults(e) {
   const number = e.target.value
-  this.setState( {number: number});
+  this.setState( {results: number});
 }
 
-handleData(e) {
+updateMovie(e) {
   const movieData = e.Search; // Films as an Array of key-value pairs [ {}, {}, {}] 
   this.setState({
     movie: movieData,
     searchStatus: true
   });
 }  
 
-handleSort(e) {
-  console.log(this.state.sort)
+updateSortCriteria(e) {
   const sortCriteria = e.target.value;
-  console.log(sortCriteria);
   this.setState({sort: sortCriteria})
-  console.log(this.state.sort); // Our problem is the update state is a bit laggy. 
 }
 
+fetchRequest() { 
```

Comment:
As `App.js` is going to grow in size quite a bit, it might be a good time to extract this function out to its own file. For example, something named `MovieRequest.js` would be fine, and would only require a refactor to convert `this.state.keyword` into a parameter of the function.